dagobah/core/core.py

```diff
@@ -690,6 +738,22 @@ def _serialize(self, include_run_logs=False, strict_json=False):
             result = json.loads(json.dumps(result, cls=StrictJSONEncoder))
         return result
 
+    def implements_expandable(obj):
+        """ Checks for methods required to expand a task """
+        for expected_method in ['expand']:
+            if not (hasattr(obj, expected_method) and
+                    callable(getattr(obj, expected_method))):
+                return False
+        return True
+
+    def implements_runnable(obj):
+        """ Checks methods required to run a task. More methods to be added """
+        for expected_method in ['start']:
+            if not (hasattr(obj, expected_method) and
+                    callable(getattr(obj, expected_method))):
+                return False
+        return True
```

Comment:
Could consolidate these to one-liners which call a private method with the list of methods they're checking for as a parameter. Removes the logic duplication.